### DISCUSSION TITLE
Plugin example with authentication

### DISCUSSION THREAD

**Discussion by eeadie** [microsoft/semantic-kernel#5748] [Q&A] [answered]
Hi,

I was wondering if there was a sample plugin that showed authentication options with semantic kernel plugins? Using Chat Copilot when I enable the Jira Plugin I can see that it supports authentication with a personal access token. Is there an example of how to implement this type of authentication?

![Screenshot 2024-04-03 144933](https://github.com/microsoft/semantic-kernel/assets/23075647/5680aced-2cc3-4408-b561-3f3db5610d0c)

**Comment by sophialagerkranspandey** [comment]
In Semantic Kernel, we take a hands-off approach to authentication as long as you're able to inject your auth key in the header.
We provide call backs that allow you to injection authentication in.

**Reply by eeadie** [reply]
Hi @sophialagerkranspandey 

Thank you for your response. My question really related to if there was a sample plugin that showed an authentication approach. All the sample plugins I have looked at (e.g. Math Plugin, ToDo Plugin) use no auth.

If the answer is no do you know if the semantic kernel team are planning to create a new sample plugin that shows an authentication approach (e.g. Personal Access Token)?

Thanks,

Euan

**Comment by dmytrostruk** [comment, marked as answer]
@eeadie Here is .NET example for Jira Plugin with authentication:
https://github.com/microsoft/semantic-kernel/blob/0a9e74a3c62aba1115bdf39c22dba669d21e7839/dotnet/samples/KernelSyntaxExamples/Example24_OpenApiPlugin_Jira.cs

In this file you can also find examples of different auth providers:
https://github.com/microsoft/semantic-kernel/blob/0a9e74a3c62aba1115bdf39c22dba669d21e7839/dotnet/samples/KernelSyntaxExamples/Example24_OpenApiPlugin_Jira.cs#L116-L122

Let me know if that helps, thank you!

**Reply by eeadie** [reply]
@dmytrostruk Thank you for your response. I have managed to get this working using a console app with the authentication details available in the Headers of the request. 

For using the plugin in Chat Copilot are you able to point me to any examples that show what the auth type can be set to? Specifically resulting in the config settings available when enabling the Jira plugin?

![Screenshot 2024-04-09 142405](https://github.com/microsoft/semantic-kernel/assets/23075647/6fa20faf-1d78-49a6-9077-b18fc12037f1)

Many Thanks,
Euan

**Reply by dmytrostruk** [reply]
@eeadie Based on the code, I believe it should be either `bearer` or `basic`:
https://github.com/microsoft/chat-copilot/blob/b0fe77ae2f9b71e5c2b529e5bd7288e3b0123c05/plugins/shared/PluginManifest.cs#L71-L74

Could you please try these values and see if that works correctly? Thanks a lot!

**Reply by eeadie** [reply]
@dmytrostruk unfortunatley neither of those options are working. Results in an error could not validate OpenAI manifest:

![Screenshot 2024-04-10 095703](https://github.com/microsoft/semantic-kernel/assets/23075647/eb442beb-3d9c-4fff-9ede-66134a0a6a37)

**Reply by dmytrostruk** [reply]
@glahaye Could you please assist with this problem?
